mlflow/models/signature.py

```diff
@@ -94,7 +96,7 @@ def __repr__(self) -> str:
 
 
 def infer_signature(
-    model_input: Any, model_output: MlflowInferableDataset = None
+    model_input: Any, model_output: "MlflowInferableDataset" = None
```

Comment:
Used a [forward-reference](https://www.python.org/dev/peps/pep-0484/#forward-references) here to avoid `MlflowInferableDataset is not defined`